File: Segunda Etapa/Semana 7/QueueTest/QueuesTest/ViewController.m
//
//  ViewController.m
//  QueuesTest
//
//  Created by Diego Vidal on 25/02/15.
//  Copyright (c) 2015 Diego Vidal. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    NSLog(@"Preparing to run code in secondary thread...");
    //dispatch_queue_t myQueue = dispatch_queue_create("My Queue", DISPATCH_QUEUE_SERIAL);
    dispatch_queue_t myQueue = dispatch_queue_create("My Queue", DISPATCH_QUEUE_CONCURRENT);
    dispatch_async(myQueue, ^{
        NSLog(@"Running code in secondary thread...");
        int value = 0;
        for (int i=0; i<100; i++) {
            for (int j=0; j<100; j++) {
                for (int n=0; n<100; n++) {
                    value += j;
                }
            }
        }
        NSLog(@"From secondary thread: value = %d", value);
    });
//    dispatch_async(myQueue, ^{
//        NSLog(@"3rd Thread...");
//        for (int k=0; k<5; k++) {
//            for (int y=0; y<250000; y++) {}
//            NSLog(@"%d\n", k);
//        }
//        NSLog(@"3rd Thread finished");
//    });
    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0),^{
        NSLog(@"3rd Thread...");
        for (int k=0; k<5; k++) {
            for (int y=0; y<250000; y++) { }
            NSLog(@"%d\n", k);
        }
        NSLog(@"3rd Thread finished");
    });
    NSLog(@"This is main thread again...");

}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: Quarta etapa/Semana 2/CoreMotionTest/core-motion/ViewController.swift
//
//  ViewController.swift
//  core-motion
//
//  Created by Diego Vidal on 26/05/15.
//  Copyright (c) 2015 Diego Vidal. All rights reserved.
//

import UIKit
import CoreMotion

class ViewController: UIViewController {

    @IBOutlet weak var lblEixoX: UILabel!
    @IBOutlet weak var lblEixoY: UILabel!
    @IBOutlet weak var lblEixoZ: UILabel!
    
    @IBOutlet weak var myImage: UIImageView!
    // Passo 1 - Instanciando o sensor, obs: boa prática utilizar singleton
    let manager = CMMotionManager()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        // Passo 2 - Verificando disponibilidade do sensor
        if manager.deviceMotionAvailable{
            
            // Passo 3 - setando o intervalo
            manager.deviceMotionUpdateInterval = 1.0/2.0
            
            // Passo 4 - startar leitura dos dados do sensor
            manager.startDeviceMotionUpdatesToQueue(NSOperationQueue.mainQueue(), withHandler: { (deviceMotion: CMDeviceMotion!, error: NSError!) -> Void in
                
                //Ler dados do sensor
                self.atualizarLabelsAndImage(deviceMotion)
                
            })
        }
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    
    func atualizarLabelsAndImage(deviceMotion: CMDeviceMotion){
        lblEixoX.text = String(format: "%.4f", deviceMotion.gravity.x)
        lblEixoY.text = String(format: "%.4f", deviceMotion.gravity.y)
        lblEixoZ.text = String(format: "%.4f", deviceMotion.gravity.z)
        
        if deviceMotion.gravity.x >= 0.07{
            myImage.image = UIImage(named: "seta_direita")
        }
        else if deviceMotion.gravity.x <= -0.07{
            myImage.image = UIImage(named: "seta_esquerda")
        }
        else if deviceMotion.gravity.y >= 0.07{
            myImage.image = UIImage(named: "seta_cima")
        }
        else if deviceMotion.gravity.y <= -0.07{
            myImage.image = UIImage(named: "seta_baixo")
        }
        else{
            myImage.image = UIImage(named: "bola_preta")
        }
        
        
    }

}

